app/models/policy.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+# Compliance policy
+class Policy < ApplicationRecord
+  DEFAULT_COMPLIANCE_THRESHOLD = 100.0
+  PROFILE_ATTRS = %w[name description account_id compliance_threshold
+                     business_objective_id].freeze
+
+  has_many :profiles, dependent: :destroy, inverse_of: :policy_object
+  has_many :benchmarks, through: :profiles
+  has_many :test_results, through: :profiles, dependent: :destroy
+
+  has_many :policy_hosts, dependent: :destroy
+  has_many :hosts, through: :policy_hosts, source: :host
+
+  belongs_to :business_objective, optional: true
+  belongs_to :account
+
+  validates :compliance_threshold, numericality: true
+  validates :account, presence: true
+  validates :name, presence: true, uniqueness: true
+
+  after_destroy :destroy_orphaned_business_objective
+  after_update :destroy_orphaned_business_objective
+  after_rollback :destroy_orphaned_business_objective
```

Comment:
Observation: These hooks might start removing b. objectives that are assigned to a Profile.

Reply:
Yes, we want that. If a BO exists on only one profile and the profile is destroyed, the BO should also be destroyed

Reply:
Yes, but not until the Policy is used. It just might get confusing if the BOs will start disappearing when testing/playing with it. Feel free to keep it.

Reply:
I'll remove it for now.

Reply:
Ok, updated. The tests for this call the method directly instead of relying on after_*, which I've removed for now.